llama/model_cn_comment.py

```diff
@@ -138,25 +157,37 @@ def apply_rotary_emb(
     xk: torch.Tensor,
     freqs_cis: torch.Tensor,
 ) -> Tuple[torch.Tensor, torch.Tensor]:
-    """Apply rotary embeddings to input tensors using the given frequency tensor.
+    """使用给定的频率张量对输入张量应用 RoPE (旋转位置编码). 参考：docs/CN/RoPE.
 
-    This function applies rotary embeddings to the given query 'xq' and key 'xk' tensors using the
-    provided frequency tensor 'freqs_cis'. The input tensors are reshaped as complex numbers, and
-    the frequency tensor is reshaped for broadcasting compatibility. The resulting tensors contain
-    rotary embeddings and are returned as real tensors.
+    本函数使用提供的频率张量 freqs_cis 对 query 张量 xq 和 key 张量 xk 应用 RoPE.
+    输入张量被 reshape 为复数形式, 频率张量也被 reshape 以确保广播兼容性.
+    本函数以实数张量的形式返回应用 RoPE 后的 query 张量和 key 张量.
 
     Args:
-        xq (torch.Tensor): Query tensor to apply rotary embeddings.
-        xk (torch.Tensor): Key tensor to apply rotary embeddings.
-        freqs_cis (torch.Tensor): Precomputed frequency tensor for complex exponentials.
+        xq (torch.Tensor): 要应用 RoPE 的 query 张量. Shape: (batch_size, n, n_heads, dim).
+        xk (torch.Tensor): 要应用 RoPE 的 key 张量. Shape: (batch_size, n, n_heads, dim).
+        freqs_cis (torch.Tensor): 预计算的复指数频率张量. Shape: (n, dim / 2).
 
     Returns:
-        Tuple[torch.Tensor, torch.Tensor]: Tuple of modified query tensor and key tensor with
-            rotary embeddings.
+        Tuple[torch.Tensor, torch.Tensor]: 包含应用 RoPE 后的 query 张量和 key 张量的元组.
     """
+    # 首先将 query 张量 reshape 为 (batch_size, n, n_heads, dim / 2, 2) 再将其转化为复数表示.
+    # 即对张量的元素两两一组分组, 以复数形式表示一组二维向量.
+    # [Shape] xq_: (batch_size, n, n_heads, dim / 2)
     xq_ = torch.view_as_complex(xq.float().reshape(*xq.shape[:-1], -1, 2))
+    # 对 key 张量执行相同操作.
+    # [Shape] xk_: (batch_size, n, n_heads, dim / 2)
     xk_ = torch.view_as_complex(xk.float().reshape(*xk.shape[:-1], -1, 2))
+    # Reshape 频率张量 freq_cis 以便广播.
+    # [Shape] freqs_cis: (1, n, 1, dim / 2)
```

Comment:
This should show the changes in `freqs_cis` shape.
```suggestion
    # [Shape] freqs_cis: (n, dim / 2) -> (1, n, 1, dim / 2)
```